src/Adapter/LinearRegressionAdapter.php

```diff
@@ -2,6 +2,102 @@
 
 namespace Zeeml\Algorithms\Adapter;
 
+use Zeeml\Algorithms\Dataset\DatasetInterface;
+
 class LinearRegressionAdapter extends AbstractAdapter
 {
+    protected $intercept;
+    protected $slope;
+
+    /**
+     * LinearRegressionAdapter constructor.
+     * @param DatasetInterface $dataset
+     */
+    public function __construct(array $data)
```

Comment:
The object should be an agnostic dataset() instance. Imagine a CsvProcessor() that would generate it. Any other processor would deliver a similar object...